braindecode/models/eegitnet.py

```diff
@@ -0,0 +1,313 @@
+# Authors: Ghaith Bouallegue <ghaithbouallegue@gmail.com>
+#
+# License: BSD-3
+import torch
+from torch import nn
+
+
+class DepthwiseConv2d(torch.nn.Conv2d):
+    def __init__(
+            self,
+            in_channels,
+            depth_multiplier=2,
+            kernel_size=3,
+            stride=1,
+            padding=0,
+            dilation=1,
+            bias=True,
+            padding_mode="zeros",
+    ):
+        out_channels = in_channels * depth_multiplier
+        super().__init__(
+            in_channels=in_channels,
+            out_channels=out_channels,
+            kernel_size=kernel_size,
+            stride=stride,
+            padding=padding,
+            dilation=dilation,
+            groups=in_channels,
+            bias=bias,
+            padding_mode=padding_mode,
+        )
+
+
+class EEG_ITNET(nn.Module):
+    """ EEG-ITNet: An Explainable Inception Temporal
+     Convolutional Network for motor imagery classification from
+     Salami et. al 2022.
+
+    See [Salami2022]_ for details.
+
+    Code adapted from https://github.com/abbassalami/eeg-itnet
+
+    Parameters
+    ----------
+    n_classes: int
+        number of outputs of the decoding task (for example number of classes in
+        classification)
+    n_in_chans: int
+        number of input EEG channels
+    input_window_samples : int
+        Number of time samples.
+    drop_prob: float
+        Dropout probability.
+
+    References
+    ----------
+    .. [Salami2022] A. Salami, J. Andreu-Perez and H. Gillmeister, "EEG-ITNet: An Explainable
+    Inception Temporal Convolutional Network for motor imagery classification," in IEEE Access,
+    doi: 10.1109/ACCESS.2022.3161489.
+    """
+
+    def __init__(self, n_classes, in_channels, input_window_samples,
+                 drop_prob=0.4):
+        super().__init__()
+        self.block11 = nn.Sequential(
```

Comment:
Inception block can be implemented as an independent `_InceptionBlock` class that implements the concatenation in the forward.